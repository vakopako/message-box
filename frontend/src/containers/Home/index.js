import React from "react";
import styled from "styled-components";

import QrCode from "../../components/QrCode";

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Title = styled.h1`
  text-align: center;
`;

const Home = () => {
  return (
    <Wrapper>
      <Title>Scan the QR code</Title>
      <QrCode />
    </Wrapper>
  );
}

export default Home;