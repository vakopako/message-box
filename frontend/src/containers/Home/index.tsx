import React from "react";
import styled from "styled-components";
import { Link } from 'react-router-dom';
import { Box, Typography, Link as MuiLink } from '@mui/material';

import QrCode from '../../components/QrCode';

const FlexBox = styled(Box)`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Heading = styled(Typography)`
  text-align: center;
`;

const HARDCODED_MESSAGE_GROUP_ID = 'c0b2b020-6787-11ee-8c99-0242ac120002';

const Home = () => {
  return (
    <FlexBox>
      <Heading variant="h2">Scan the QR code</Heading>
      <QrCode groupId={HARDCODED_MESSAGE_GROUP_ID} />
      <MuiLink component={Link} to={`/${HARDCODED_MESSAGE_GROUP_ID}`}>
        or just click here
      </MuiLink>
    </FlexBox>
  );
};

export default Home;