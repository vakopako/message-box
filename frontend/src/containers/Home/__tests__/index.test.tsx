import React from "react";
import { describe, it, expect } from '@jest/globals';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

import Home from '../index';

describe('Home', () => {
  it('should render the Home component according to the snapshot', () => {

    const componentTree = (
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );

    const tree = renderer.create(componentTree).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
