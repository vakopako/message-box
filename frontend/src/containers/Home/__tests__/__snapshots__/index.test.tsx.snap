// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render the Home component according to the snapshot 1`] = `
<div
  className="sc-bdfCDU eMheSe MuiBox-root css-0"
>
  <h2
    className="MuiTypography-root MuiTypography-h2 sc-gsTDqH bRtRJy css-1sra7t5-MuiTypography-root"
  >
    Scan the QR code
  </h2>
  <canvas
    role="img"
  />
  <a
    className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
    href="/c0b2b020-6787-11ee-8c99-0242ac120002"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
  >
    or just click here
  </a>
</div>
`;
